File: src/main/java/beeroclock/BeerOClockChecker.java
package beeroclock;

import java.util.Calendar;

public class BeerOClockChecker {

    private final int BEER_O_CLOCK = 16;

    public boolean isBeerOClock() {
        return now().get(Calendar.HOUR_OF_DAY) >= BEER_O_CLOCK;
    }

    protected Calendar now() {
        return Calendar.getInstance();
    }
}

File: csharp/TestingExercises/TestingExercises/src/BeerOClockChecker.cs
using System;

namespace TestingExercises
{
    public class BeerOClockChecker
    {
        private const int BeerOClock = 16;

        public bool IsBeerOClock()
        {
            return Now().Hour >= BeerOClock;
        }

        protected virtual DateTime Now()
        {
            return DateTime.Now;
        }
    }
}